### PR TITLE
Logging additional details when fulfillment fails

### PR DESCRIPTION
**Pull request by clintonb**
@rlucioni @dianakhuang @stephensanchez

### CODE REVIEW

**Comment by rlucioni** [comment]
:+1:

Don't forget to update the ecommerce service to install this new version.

**Comment by clintonb** [comment]
Will do. Waiting for the build to finish before merging.

File: extensions/fulfillment/modules.py [hunk 1]
```diff
@@ -182,8 +182,15 @@ def fulfill_product(self, order, lines):
                     logger.info("Success fulfilling line [%d] of order [%s].", line.id, order.number)
                     line.set_status(LINE.COMPLETE)
                 else:
+                    try:
+                        data = response.json()
+                        reason = data.get('message')
+                    except Exception:   # pylint: disable=broad-except
+                        reason = '(No detail provided.)'
```
Comment (dianakhuang) [inline extensions/fulfillment/modules.py]: Is there a particular reason why this message is in parens?

**Comment by dianakhuang** [comment]
Looks like you've angered Coveralls, but otherwise, :rocket:

**Comment by rlucioni** [comment]
I'm forced to withdraw my thumb until Coveralls is happy.

File: extensions/fulfillment/modules.py (hunk 1 above)
Comment (clintonb) [inline extensions/fulfillment/modules.py]: To indicate that the message is coming from our code, not the API.

**Comment by clintonb** [comment]
Updated.

**Comment by clintonb** [comment]
Updated.

**Comment by stephensanchez** [comment]
:+1:

**Pull request merged**